mesonbuild/modules/i18n.py

```diff
@@ -269,5 +288,65 @@ def gettext(self, state: 'ModuleState', args: T.Tuple[str], kwargs: 'Gettext') -
 
         return ModuleReturnValue([gmotargets, pottarget, updatepotarget], targets)
 
+    @FeatureNew('i18n.itstool_join', '0.61.0')
+    @noPosargs
+    @typed_kwargs(
+        'i18n.itstool_join',
+        CT_BUILD_BY_DEFAULT,
+        CT_INPUT_KW,
+        CT_INSTALL_DIR_KW,
+        CT_INSTALL_TAG_KW,
+        CT_OUTPUT_KW,
+        INSTALL_KW,
+        _ARGS.evolve(),
+        KwargInfo('its_files', ContainerTypeInfo(list, str)),
+        KwargInfo('mo_targets', ContainerTypeInfo(list, build.CustomTarget), required=True),
+    )
+    def itstool_join(self, state: 'ModuleState', args: T.List['TYPE_var'], kwargs: 'ItsJoinFile') -> ModuleReturnValue:
+        if not shutil.which('itstool'):
+            self.noitstool_error()
```

Comment:
This seems like a weird thing to delegate to a dispatch function as it is only used once, and we should really just use the find_program implementation, which respects tools defined in cross files, then pass it as an argument to the helper script.